scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,409 @@
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+        return (Serializable) filesRelativePathName;
+    }
+
+    private List<String> FtpGet(FTPClient ftpClient) throws IOException {
+        List<String> filesRelativePathName;
+        getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+        FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+        if (ftpFile.length == 0) {
+            throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+        }
+        filesRelativePathName = new ArrayList<>();
+        // If it is a single file:
+        if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+            String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+            if (downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath)) {
+                getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+                filesRelativePathName.add(saveFilePath);
+            } else {
+                throw new IOException("Error: COULD NOT download the file: " + ftpRemoteFilePath);
+            }
+
+            // If the file is a zip, and ftpExtractArchive is set to true
+            if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                extractArchive(saveFilePath);
+            }
+        } // If it is a folder, download all its contents recursively
+        else {
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            filesRelativePathName.addAll(new HashSet(downloadDirectory(ftpClient, "", "", ftpLocalFilePath)));
+        }
+        getOut().println("END Import file(s) from FTP.");
+        return filesRelativePathName;
+    }
+
+    private void extractArchive(String saveFilePath) throws IOException {
+        try (ZipFile zipFile = new ZipFile(saveFilePath)) {
+            // Create an enumeration of the entries in the zip file
+            Enumeration zipFileEntries = zipFile.entries();
+            // Process each entry
+            while (zipFileEntries.hasMoreElements()) {
+                // grab a zip file entry
+                ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                Path path = Paths.get(ftpLocalFilePath, entry.getName());
+                getOut().println("path: " + path.toString());
+                if (entry.isDirectory()) {
+                    Files.createDirectories(path);
+                } else {
+                    Path parentDir = path.getParent();
+                    getOut().println("parentDir of path: " + parentDir.toString());
+                    if (!Files.exists(parentDir)) {
+                        Files.createDirectories(parentDir);
+                    }
+                    if (Files.exists(path)) {
+                        new File(path.toString()).delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)